### PR TITLE
Laboratory work #4, Darya Balba - 21FPL2

### PR DESCRIPTION
**Pull request by daryabalba**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -2,8 +2,13 @@
 Lab 3
 Extract keywords based on TextRank algorithm
 """
+import os
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:5]: Revert to the main

File: lab_3_keywords_textrank/start.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 TextRank keyword extraction starter
 """
 
+import json
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/start.py:5]: .

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -84,32 +111,67 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
         """
         Constructs all the necessary attributes
         """
-        pass
+        # if not isinstance(stop_words, tuple) or not isinstance(punctuation, tuple):
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:114]: remove

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
What about 10?

File: lab_3_keywords_textrank/start.py (hunk 2 above)
Comment (daryabalba) [reply lab_3_keywords_textrank/start.py]: Done

File: lab_4_summarization_textrank/main.py [hunk 4]
```diff
@@ -2,15 +2,33 @@
...
+    """
+    Checks if the variable is of an apropriate type
+    param: variable
+    param: possible_var_type
+    param: container_value_type (default = None)
+    return:
+    """
+    if not isinstance(variable, possible_var_type) or isinstance(variable, bool):
+        raise ValueError
+    if container_value_type:
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:26]: if not -> return

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -2,15 +2,33 @@
...
 from lab_3_keywords_textrank.main import TextEncoder, \
-    TextPreprocessor
+    TextPreprocessor, TFIDFAdapter
+
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
 
+def check_types(variable: Any, possible_var_type: Type, container_value_type: Union[Type, None] = None) -> None:
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:16]: container_value_type: Optional[Type] = None

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/main.py (hunk 4 above)
Comment (daryabalba) [reply lab_4_summarization_textrank/main.py:26]: Why? It's an optional variable and those actions pass only if i have it

Comment (daryabalba) [reply lab_4_summarization_textrank/main.py]: I changed it, is this what you meant?

File: lab_4_summarization_textrank/main.py (hunk 5 above)
Comment (daryabalba) [reply lab_4_summarization_textrank/main.py]: Done

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -2,15 +2,35 @@
...
 
+def check_types(variable: Any, possible_var_type: Type, container_value_type: Optional[Type] = None) -> None:
+    """
+    Checks if the variable is of an appropriate type
+    param: variable
+    param: possible_var_type
+    param: container_value_type (default = None)
+    return:
+    """
+    if not isinstance(variable, possible_var_type) or isinstance(variable, bool):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:24]: .dict? bool? improve

File: lab_4_summarization_textrank/main.py [hunk 7]
```diff
@@ -120,23 +168,38 @@ def __init__(self) -> None:
...
+        check_types(sentences, tuple, Sentence)
+
+        coded_sent = []
+        for one_sentence in sentences:
+            preprocessed_tokens = one_sentence.get_preprocessed()
+            self._learn_indices(preprocessed_tokens)
+            for words in preprocessed_tokens:
+                coded_sent.append(self._word2id[words])
+            one_sentence.set_encoded(tuple(coded_sent))
+            coded_sent = []
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:202]: ?

File: lab_4_summarization_textrank/main.py [hunk 8]
```diff
@@ -146,7 +209,29 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
     :param other_sequence: a sequence of items
     :return: similarity score
     """
-    pass
+    if not sequence or not other_sequence:
+        return 0
+
+    for variable in (sequence, other_sequence):
+        if not isinstance(variable, (list, tuple)):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:216]: check_types?

File: lab_4_summarization_textrank/main.py [hunk 9]
```diff
@@ -193,15 +308,40 @@ def get_similarity_score(self, sentence: Sentence, other_sentence: Sentence) ->
...
+        return self._matrix[idx1][idx2]
 
     def fill_from_sentences(self, sentences: tuple[Sentence, ...]) -> None:
         """
         Updates graph instance with vertices and edges extracted from sentences
         :param sentences
         :return:
         """
-        pass
+        if not sentences:
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:330]: check-types

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**